File: python/MASTER_SLAVE_WHOLE_PART_PROXY/Reservacion.py
from datetime import date

class DatosCliente:
    
    nombres         : str
    apellidoPaterno : str
    apellidoMaterno : str
    edad            : int
    tipoCliente     : str
    
    def setDatosCliente(self, nombres         : str,
                              apellidoPaterno : str,
                              apellidoMaterno : str,
                              edad            : int,
                              tipoCliente      : bool) -> None:
        self.nombres         = nombres
        self.apellidoPaterno = apellidoPaterno
        self.apellidoMaterno = apellidoMaterno
        self.edad            = edad
        self.tipoCliente     = tipoCliente
    
class DatosCompra:
    
    numeroTarjeta    : str
    fechaVencimiento : str
    
    def setDatosCompra(self, numeroTarjeta    : str,
                             fechaVencimiento : str):
        self.numeroTarjeta    = numeroTarjeta
        self.fechaVencimiento = fechaVencimiento
    
class DatosFuncion:
    
    asiento  : str
    horario  : date
    pelicula : str
    
    def setDatosFuncion(self, asiento : str,
                              horario : date,
                              pelicula : str):
        self.asiento  = asiento
        self.horario  = horario
        self.pelicula = pelicula
        
class Reservacion:
    
    datosCliente = DatosCliente()
    datosCompra  = DatosCompra()
    datosFuncion = DatosFuncion()
    
    def mostrarReservacion(self):
        print(
        f'''
        Reservación:
        
        Datos cliente:
            Nombres          : {self.datosCliente.nombres}
            Apellido paterno : {self.datosCliente.apellidoPaterno}
            Apellido materno : {self.datosCliente.apellidoMaterno}
            Edad             : {self.datosCliente.edad}
            Tipo de cliente  : {self.datosCliente.tipoCliente}
            
        Datos compra:
            Número de tarjeta               : {self.datosCompra.numeroTarjeta}
            Fecha de vencimiento de tarjeta : {self.datosCompra.fechaVencimiento}
            
        Datos función:
            Asiento  : {self.datosFuncion.asiento}
            Horario  : {self.datosFuncion.horario}
            pelicula : {self.datosFuncion.pelicula}
            
        '''
        )
        
    def devolverReservacion(self):
        return (self.datosCliente,
                self.datosCompra,
                self.datosFuncion)

File: php/BUILDER_COMPOSITE/reservaciones/Reservacion.php
<?php

require_once './datos/DatosCliente.php';
require_once './datos/DatosCompra.php';
require_once './datos/DatosFuncion.php';

class Reservacion {
    private $datosCliente;
    private $datosFuncion;
    private $datosCompra;
    
    public function __construct() {
        $this->datosCliente = new DatosCliente();
        $this->datosCompra = new DatosCompra();
        $this->datosFuncion = new DatosFuncion();
    }
    
    
    public function mostrarReservacion(){
        return $this->datosCliente->getDatosCliente().$this->datosCompra->getDatosCompra().$this->datosFuncion->getDatosFuncion(); 
    }
    
    public function gestionarReservacion() {
        echo 'Gestionar Reservación';
    }
    public function getDatosCliente() {
        return $this->datosCliente;
    }

    public function getDatosFuncion() {
        return $this->datosFuncion;
    }

    public function getDatosCompra() {
        return $this->datosCompra;
    }
    public function setDatosCliente($datosCliente): void {
        $this->datosCliente = $datosCliente;
    }

    public function setDatosFuncion($datosFuncion): void {
        $this->datosFuncion = $datosFuncion;
    }

    public function setDatosCompra($datosCompra): void {
        $this->datosCompra = $datosCompra;
    }


    
    


}
